src/main/scala/commonist/task/ChangeDirectoryTask.scala

```diff
@@ -31,11 +46,14 @@ final class ChangeDirectoryTask(mainWindow:MainWindow, imageListUI:ImageListUI,
 		val	(readable,unreadable)	= sorted partition { _.canRead }
 		unreadable foreach { it => WARN("cannot read", it) }
 
-		val max		= readable.length
+		val (xmls,images) = readable partition { f => isXml(f.getName()) }
+		val oaipmh = xmls.map(XML loadFile).filter("OAI-PMH" == _.label).map(new OaiPmh2(_, getOaiPmhProps)).toList
```

Comment:
better use `.toVector`, as a general purpose collection `Vector` is usually a better choice than `List`